src/cdomain/value/cdomains/int/defExcDomain.ml

```diff
@@ -382,11 +380,11 @@ struct
   (* The equality check: *)
   let eq ik x y = match x,y with
     (* Not much to do with two exclusion sets: *)
-    | `Excluded _, `Excluded _ -> top ()
+    | `Excluded _, `Excluded _ -> top_of ik
     (* Is x equal to an exclusion set, if it is a member then NO otherwise we
      * don't know: *)
-    | `Definite x, `Excluded (s,r) -> if S.mem x s then of_bool IInt false else top ()
-    | `Excluded (s,r), `Definite x -> if S.mem x s then of_bool IInt false else top ()
+    | `Definite x, `Excluded (s,r) -> if S.mem x s then of_bool IInt false else top_of ik
+    | `Excluded (s,r), `Definite x -> if S.mem x s then of_bool IInt false else top_of ik
```

Comment:
Should this not be `top_of Iint` then? Just for symmetry with the other case?

Reply:
That sounds right, so I've changed it in `eq` and `ne`.

Although these things look like a mess: other comparisons (e.g. `lt`) go as far as returning `top_bool`, regardless of `ik`.